.github/workflows/publish.yml

```diff
@@ -22,32 +22,33 @@ jobs:
           channel: "stable"
       - name: Install dependencies
         run: flutter pub get
-      - name: Check Publish Warnings
-        run: dart pub publish --dry-run
-      # - name: Publish
-      #   uses: k-paxian/dart-package-publisher@v1.5.1
-      #   with:
-      #     credentialJson: ${{ secrets.PUB_CREDENTIALS }}
-      #     flutter: true
-      #     skipTests: true
+      - name: Setup pub credentials
+        run: |
+          mkdir -p ~/.pub-cache
+          echo '${{ secrets.PUB_CREDENTIALS }}' > ~/.pub-cache/credentials.json
+          echo "Checking credentials file exists:"
+          cat ~/.pub-cache/credentials.json
```

Comment:
dont forget to remove this before merging. github redacts it but still not good to print it out

you could do something like this instead

`[ -s ~/.pub-cache/credentials.json ] && echo credentials exists`